src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -375,30 +375,65 @@ private static bool RemovalMayIntroduceCastAmbiguity(ParenthesizedExpressionSynt
         {
             // Be careful not to break the special case around (x)(-y)
             // as defined in section 7.7.6 of the C# language specification.
+            //
+            // cases we can't remove the parens for are:
+            //
+            //      (x)(+y)
+            //      (x)(-y)
+            //      (x)(&y) // unsafe code
+            //      (x)(*y) // unsafe code
+            //
+            // Note: we can reomve the parens if the (x) part is unambiguously a type.
```

Comment:
typo: remove

Reply:
really? this doesn't seem like the original issue... I think @gafter would want to keep the parens even if it's an `int`